### ISSUE TITLE
componentWillReceiveProps is deprecated and will be removed in the next major version

### ISSUE THREAD

**Issue by swillis12** [react-component/checkbox#80]
componentWillReceiveProps needs to be removed to eliminate this warning:
```
Warning: componentWillReceiveProps is deprecated and will be removed in the next major version. Use static getDerivedStateFromProps instead.
```

Offending implementation found here:
https://github.com/react-component/checkbox/blob/acb9ef334e56ca1ee8b806deafcdb3564183c19e/src/Checkbox.jsx#L46
